### DISCUSSION TITLE
Optional Query Parameters in Quarkus Rest Client

### DISCUSSION THREAD

**Discussion by jack-farley** [quarkusio/quarkus#30831] [Q&A]
I'm using the Quarkus Reactive Rest Client to access an external api, and a few of the endpoints allow for several optional query parameters. Based on the situation, I'm going to be providing some of these parameters in many different combinations. It seems like to do this I would need a different method in my rest client for every combination of parameters that I provide. Is there a better way to do this?

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
I don't remember if `Optional<T>` works for the REST client, have you tried it?

**Reply by jack-farley** [reply]
It doesn't seem to. When I make a field marked with `@QueryParam` an `Optional<String>`, for example, the query will look something like `https://url.com/path?test_param=Optional%5Btest%5D`.

**Reply by jack-farley** [reply]
It would be a cool feature if optional parameters and headers were left out when empty and included as regular values when not empty.

**Reply by geoand** [reply]
I actually that they were. I'll have to take a look

**Comment by geoand** [comment]
We support using a Map of MultivaluedMap with @RestQuery.

Have you tried that?

**Reply by jack-farley** [reply]
That looks like it'll do the job, thanks!

**Comment by jack-farley** [comment]
@geoand I'm now running into a similar problem with Java objects that are being serialized into JSON objects in the bodies of POST requests. If I have several combinations of fields that might need to be in a JSON object, would I need a separate Java object for each combo, or do you know of a workaround?

**Reply by jack-farley** [reply]
The main problem is that fields I am assigning null are appearing in the JSON body, but I need them to be omitted from the JSON object.

**Reply by geoand** [reply]
Have you tried annotating with `@JsonInclude(Include.NON_NULL)`?

**Reply by jack-farley** [reply]
That works! Thanks for your help!

**Reply by geoand** [reply]
👌

**Reply by cvgaviao** [reply]
Hey @jack-farley , could you please show me an example of a URL you resolved and its method signature ?
